### PR TITLE
feat: preview r code in connect to read pin

### PR DESCRIPTION
**Pull request by isabelizimm**
parity to https://github.com/rstudio/pins-r/pull/806/files

on preview in Posit Connect, users should be able to see how to load the data in R and Python if applicable.

### CODE REVIEW

**Comment by isabelizimm** [comment]
View it on Connect: https://colorado.posit.co/rsc/content/4a580f0f-91f2-40cc-a060-d2e2a5b090a4/

File: pins/boards.py [hunk 1]
```diff
@@ -1123,6 +1123,12 @@ def prepare_pin_version(self, pin_dir_path, x, name: "str | None", *args, **kwar
             # TODO(compat): set display none in index.html
             context["data_preview"] = json.dumps({})
 
+        # do not show r code if not round-trip friendly
+        if meta.type in ["joblib"]:
```
Comment (isabelizimm) [inline pins/boards.py:1127]: type options are "csv", "arrow", "parquet", "joblib", or "json". we will hide the R code if the pin is of `joblib` type

File: pins/rsconnect/html/index.html [hunk 2]
```diff
@@ -50,15 +50,27 @@ <h3>{{pin_name}}</h3>
...
       hljs.initHighlightingOnLoad();
     </script>
     </section>
 
+    <section style="{{ show_r_style }}">
+      <h3>R Code</h3>
+        <pre id="pin-r" class="pin-code"><code class="r">library(pins)
+
+  board <- {{board_deparse}}
+  pin_read(board, "{{pin_name}}")</code></pre>
```
Comment (juliasilge) [inline pins/rsconnect/html/index.html:71]:
```suggestion
board <- {{board_deparse}}
pin_read(board, "{{pin_name}}")</code></pre>
```

Comment (juliasilge) [reply pins/rsconnect/html/index.html:71]: We can't use the same `board_deparse` object here, right? It needs to be something R specific? I believe it should be `"board_connect(auth = "envvar")"`.

**Review by juliasilge — CHANGES_REQUESTED** [review]
Looking good!

File: pins/rsconnect/html/index.html (hunk 2 above)
Comment (isabelizimm) [reply pins/rsconnect/html/index.html]: ah, yep! if we wanted, we could deparse to be `board_connect()`, but the preview adds the raw `server_url` for Python. I've changed it to `board_connect(auth = "envvar")`

**Review by juliasilge — APPROVED** [review]

**Pull request merged**